From 41811d473df9a1cb09bc4f44eba3147dcec231b1 Mon Sep 17 00:00:00 2001
From: Dietmar Maurer <dietmar@proxmox.com>
Date: Wed, 8 May 2013 10:01:29 +0200
Subject: [PATCH v5 9/9] vma: only store the basename of a configuration file

It makes no sense to store the whole filesystem path here.

Signed-off-by: Dietmar Maurer <dietmar@proxmox.com>
---
 vma-writer.c |    5 ++++-
 vma.c        |    2 +-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/vma-writer.c b/vma-writer.c
index 9228ca6..3cc9cf2 100644
--- a/vma-writer.c
+++ b/vma-writer.c
@@ -145,7 +145,10 @@ int vma_writer_add_config(VmaWriter *vmaw, const char *name, gpointer data,
     assert(data);
     assert(len);
 
-    uint32_t name_ptr = allocate_header_string(vmaw, name);
+    gchar *basename = g_path_get_basename(name);
+    uint32_t name_ptr = allocate_header_string(vmaw, basename);
+    g_free(basename);
+
     if (!name_ptr) {
         return -1;
     }
diff --git a/vma.c b/vma.c
index bcde379..f4a8218 100644
--- a/vma.c
+++ b/vma.c
@@ -33,7 +33,7 @@ static void help(void)
         "\n"
         "vma list <filename>\n"
         "vma create <filename> [-c config] <archive> pathname ...\n"
-        "vma extract <filename> [-v] [-r] <targetdir>\n"
+        "vma extract <filename> [-v] [-r <fifo>] <targetdir>\n"
         "vma verify <filename> [-v]\n"
         ;
 
-- 
1.7.10.4

